twillio.go

```
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+func mustHaveEnv(n string) (string, error) {
+	if v, ok := os.LookupEnv(n); ok {
+		log.Printf("%s: %s\n", n, v)
+		return v, nil
+	}
+	return "", fmt.Errorf("%s not set", n)
+}
+
+// Texter keeps track of Twillio account info
+type Texter struct {
+	accountSID    string
+	authToken     string
+	sendingNumber string
+	url           string
+}
+
+// NewTexter creates a new Texter from using environment variables, or errors
+func NewTexter() (*Texter, error) {
+	var s, t, n string
+	var err error
+	if s, err = mustHaveEnv("TWILLIO_ACCOUNT_SID"); err != nil {
+		return nil, err
+	}
+	if t, err = mustHaveEnv("TWILLIO_AUTH_TOKEN"); err != nil {
+		return nil, err
+	}
+	if n, err = mustHaveEnv("TWILLIO_SENDING_NUMBER"); err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("https://api.twilio.com/2010-04-01/Accounts/%s/Messages.json", s)
+
+	return &Texter{
+		accountSID:    s,
+		authToken:     t,
+		sendingNumber: n,
+		url:           u,
+	}, nil
+}
+
+// SendText sends the message to the destination number
+func (t *Texter) SendText(destination string, message string) error {
+	msgData := url.Values{}
+	msgData.Set("To", destination)
+	msgData.Set("From", t.sendingNumber)
+	msgData.Set("Body", message)
+	msgDataReader := *strings.NewReader(msgData.Encode())
+	client := &http.Client{}
+	req, _ := http.NewRequest("POST", t.url, &msgDataReader)
```

Why are errors swallowed here but raised elsewhere? Is the assumption that `t.url` and `&msgDataReader` will _always_ be valid in the eyes of the `http` library?

---

That was copy pasta code - I hate doing that and should fix it.